homeassistant/components/upb/__init__.py

```diff
@@ -0,0 +1,126 @@
+"""Support the UPB PIM."""
+import asyncio
+import logging
+
+import upb_lib
+
+from homeassistant.const import CONF_FILE_PATH, CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+
+UPB_PLATFORMS = ["light"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, hass_config: ConfigType) -> bool:
+    """Set up the UPB platform."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up a new config_entry for UPB PIM."""
+
+    url = config_entry.data[CONF_HOST]
+    file = config_entry.data[CONF_FILE_PATH]
+
+    upb = upb_lib.UpbPim({"url": url, "UPStartExportFile": file})
+    upb.connect()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = {"upb": upb}
+
+    for component in UPB_PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload the config_entry."""
+
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, component)
+                for component in UPB_PLATFORMS
+            ]
+        )
+    )
+
+    if unload_ok:
+        upb = hass.data[DOMAIN][config_entry.entry_id]["upb"]
+        upb.disconnect()
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+
+    return unload_ok
+
+
+class UpbEntity(Entity):
+    """Base class for all UPB entities."""
+
+    def __init__(self, element, unique_id, upb):
+        """Initialize the base of all UPB devices."""
+        self._upb = upb
+        self._element = element
+        self._unique_id = (
+            f"{unique_id}_{self.__class__.__name__.lower()}_{element.addr}"
```

Comment:
Can you use something else besides the class name here?   I've seen unexpected breakage in the past because a class was renamed and the name being used was missed which would cause the unique id to be unstable in this case.